### ISSUE TITLE
Zoomtext reads out all radio button options

### ISSUE THREAD

**Issue by dashouse** [alphagov/govuk-frontend#1598]
Submitted by user:

> ...when I use the 'tab' key on the keyboard to navigate to these elements, once I have selected the yes radio button for example, the software reads the yes radio button as 'yes yes' (so reads it twice). Then for the no radio button it reads 'no yes'.

They are using Zoomtext 10.11 or later on IE11 and have confirmed it also happens when using the example from the Design System - https://design-system.service.gov.uk/components/radios/default/index.html

**Comment by selfthinker** [comment]
Without further testing (yet) I'm not sure if this is just the way ZoomText handles radio buttons. I would recommend to try it out with native radio buttons to see if the issue still exists.
If yes, it’s a thing (it might not even be a bug) in ZoomText which we could report to them. If no, it would need some investigating on our side what is causing it.

It would also be good to know which ZoomText they are using. Is it "Fusion Professional" or "ZoomText Magnifier/Reader"?

**Comment by dashouse** [comment]
The user is using "ZoomText Magnifier/Reader" and this is not happening on stripped back HTML radios or with radios on other sites (e.g. https://www.thetrainline.com/) so it sounds like it's something to do with our markup specifically.

**Comment by dashouse** [comment]
A second Radio Buttons ZoomText issue from the same user

_When using the AppReader part of Zoomtext and as it reads through the screen. The software reads out ‘Radio button 1 label’ for the first radio button which is correct, however then reads out ‘Radio button 1 label’ again for the second radio button when it should read out ‘Radio button 2 label’. The software also doesn’t read out the hint text underneath each radio button_

**Comment by 36degrees** [comment]
> _When using the AppReader part of Zoomtext and as it reads through the screen. The software reads out {{Radio button 1 label}} for the first radio button which is correct, however then reads out {{Radio button 1 label}} again for the second radio button when it should read out {{Radio button 2 label}}. The software also doesn’t read out the hint text underneath each radio button_

That sounds a lot like it could be related to https://github.com/alphagov/reported-bugs/issues/2

**Comment by nickcolley** [comment]
We want to do an investigation to check if our radio buttons are consistent with the behaviour of regular radio buttons.

2 hours timebox

**Comment by grahamadam84** [comment]
> The user is using "ZoomText Magnifier/Reader" and this is not happening on stripped back HTML radios or with radios on other sites (e.g. https://www.thetrainline.com/) so it sounds like it's something to do with our markup specifically.

Hi @dashouse, I realise this was some time ago but just hoping you can remember if the when ZoomText reads the trainline radios it using the value text or the span text?

**Comment by 36degrees** [comment]
Putting this in upcoming sprints again. If we can't reproduce it in a couple of hours, let's close this.

**Comment by htmlandbacon** [comment]
Just wondering how you got on with this @36degrees ?

**Comment by 36degrees** [comment]
I can see this happening with Firefox 84 and ZoomText – using the ['basic' example on MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio#Defining_a_radio_group), but I still can't reproduce the issue with Internet Explorer 11.

https://user-images.githubusercontent.com/121939/102899045-4761af00-4462-11eb-9bf3-5d0c4ed31548.mov

**Comment by 36degrees** [comment]
I've raised this as a bug – however, from further testing it doesn't seem to affect the radios in the Design System (inline or normal)

https://user-images.githubusercontent.com/121939/102902561-120b9000-4467-11eb-94b9-1320fdbfa1f3.mov

**Comment by 36degrees** [comment]
Closing this as we've reported the bug to ZoomText and it doesn't appear to affect our radio or checkbox components anyway for some reason (though it's hard to test with the examples in the review app or Design System because of https://github.com/alphagov/reported-bugs/issues/2, I've tested a few other pages in ZoomText with IE11 and they behave as expected)
